buidl/script.py

```diff
@@ -54,9 +54,12 @@ def __add__(self, other):
         return Script(self.commands + other.commands)
 
     @classmethod
-    def parse(cls, stream):
+    def parse(cls, stream, is_p2sh_multisig=False):
         # get the length of the entire field
-        raw = read_varstr(stream)
+        if is_p2sh_multisig:
+            raw = stream.read()
```

Comment:
This isn't good. You'll consume the entire stream. This could be the rest of the transaction, for example, though for redeem script, I think it gets put into a bytestream anyway.